### PR TITLE
Added Ruby string interpolation

### PR DESCRIPTION
**Pull request by westonganger**
The only case it doesnt handle is if the string interpolation is not inside a string

### CODE REVIEW

File: components/prism-ruby.js [hunk 1]
```diff
@@ -5,17 +5,33 @@
  * 		constant, builtin, variable, symbol, regex
  */
 Prism.languages.ruby = Prism.languages.extend('clike', {
-	'comment': /#[^\r\n]*(\r?\n|$)/,
-	'keyword': /\b(alias|and|BEGIN|begin|break|case|class|def|define_method|defined|do|each|else|elsif|END|end|ensure|false|for|if|in|module|new|next|nil|not|or|raise|redo|require|rescue|retry|return|self|super|then|throw|true|undef|unless|until|when|while|yield)\b/,
+        'comment': /#(?!\{[^\r\n]*?\})[^\r\n]*(\r?\n|$)/g,
```
Comment (Golmote) [inline components/prism-ruby.js]: Please use tabs to indent, not spaces.

Also, why the `g` flag everywhere? It is not used by Prism.

File: components/prism-ruby.js [hunk 2]
```diff
@@ -5,17 +5,33 @@
...
+        pattern: /("|')(\\\n|\\?.)*?\1/,
+        inside: {
+          'interpolation': {
+            pattern: /#\{[^}]+\}/,
+            inside: {
+              'delimiter': {
+                pattern: /^#\{|\}$/,
+                alias: 'tag'
+              },
+              rest: Prism.languages.extend('ruby')
```
Comment (Golmote) [inline components/prism-ruby.js]: You should use `Prism.util.clone` instead of `extend`.

File: components/prism-ruby.js [hunk 3]
```diff
@@ -5,17 +5,33 @@
...
 		lookbehind: true
 	},
-	'variable': /[@$]+\b[a-zA-Z_][a-zA-Z_0-9]*[?!]?\b/,
-	'symbol': /:\b[a-zA-Z_][a-zA-Z_0-9]*[?!]?\b/
+	'variable': /[@$]+\b[a-zA-Z_][a-zA-Z_0-9]*[?!]?\b/g,
+	'symbol': /:\b[a-zA-Z_][a-zA-Z_0-9]*[?!]?\b/g
 });
+
+Prism.languages.ruby.string = {
+        pattern: /("|')(\\\n|\\?.)*?\1/,
```
Comment (Golmote) [inline components/prism-ruby.js]: Again, be careful with the indentation. Prism uses tabs, not spaces.

**Comment by Golmote** [comment]
I don't have time to test it right now, but I quickly reviewed the code. Please take a look at my comments ;)

**Comment by Golmote** [comment]
I finally got some time to play around with your PR.
I think you could fix the second failure ("Double quoted strings inside of string interpolation") by changing the `string` pattern to `/("|')(#\{[^}]+\}|\\\n|\\?.)*?\1/`. What do you think?

**Comment by LeaVerou** [comment]
@Golmote do you think this is good to merge?

**Comment by Golmote** [comment]
Oh, I did not notice the last commit. So yes, it looks good to me now.

**Pull request merged**